api/v1alpha1/api.go

```diff
@@ -144,3 +150,65 @@ const (
 	// This can be used to describe the routing behavior in HTTPRoute referenced by LLMRoute.
 	LLMModelHeaderKey = "x-envoy-ai-gateway-llm-model"
 )
+
+// LLMProviderType specifies the type of the LLMSecurityPolicy.
+type LLMProviderType string
+
+const (
+	LLMProviderTypeAPIKey LLMProviderType = "APIKey"
+)
+
+// +kubebuilder:object:root=true
+
+// LLMSecurityPolicy specifies the provider specific configuration like authorization, JWT, and API Key.
+type LLMSecurityPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              LLMSecurityPolicySpec `json:"spec,omitempty"`
+}
+
+// LLMSecurityPolicySpec specifies a provider (e.g.AWS Bedrock, Azure etc.) specific-configuration like auth
+type LLMSecurityPolicySpec struct {
+	// Type specifies the type of the provider. Currently, only "APIKey" and "AWSBedrock" are supported.
+	//
+	// +kubebuilder:validation:Enum=APIKey;AWSBedrock
+	Type LLMProviderType `json:"type"`
```

Comment:
I think this is not provider type, it is the authentication mechanism with APIKey or AWS Auth.
"API Key" is a simple authentication method used to grant access to an API, while "AWS Auth" refers to the broader system of authentication within Amazon Web Services (AWS)

Reply:
maybe the envoy gateway SecurityPolicy is already good enough, they just do not support AWS Auth
https://gateway.envoyproxy.io/contributions/design/security-policy/#features--api-fields

Reply:
I think this is for LLM Backend authentication with the provider, and EG SecurityPolicy is used to authenticate the clients/users.